### DISCUSSION TITLE
Servlet in client mode

### DISCUSSION THREAD

**Discussion by ghost** [phax/phase4#216] [Q&A]
We are testing phase4 as a client. 

We are wondering why is this necessary ? WebScopeManager.onGlobalBegin (MockServletContext.create ());
Because we won't be in a web application with servlet to send the messages. 

Is this still necessary ? If yes, I imagine we would need to include this dependency as in the phase4 client example not in provided scope ? 
```

<dependency>
      <groupId>jakarta.servlet</groupId>
      <artifactId>jakarta.servlet-api</artifactId>
</dependency>
```

**Comment by phax** [comment]
As part of the surrounding framework, it contains elements called "scopes". A scope basically defines the lifetime of its contained objects. This also applies to "global variables". By having a scope that is created and deleted, also "global" variables can be safely cleansed etc.
This pattern is especially helpful in testing, when you can make sure, each test starts with a clean "global scope".

So yes, you need to include the `servlet-api` into your dependencies "unconditionally", In `phase4-lib` it is set with scope `provided` as application servers usually contain the dependency.
